### ISSUE TITLE
This Parameter leads to Missing semicolon error

### ISSUE THREAD

**Issue by sandrozbinden** [palantir/tslint#2739]
### Bug Report

- __TSLint version__: 5.0.0
- __TypeScript version__: 2.1.6
- __Running TSLint via__: grunt-tslint

#### TypeScript code being linted

```ts
  findByName() {
    let arr = ['a', 'b', 'c'];
    arr.find(this.isName, 'b');
  }

  isName(this: string, name: string): boolean {
    return this === name;
  }
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "extends": [
        "tslint-ionic-rules"
    ],
    "jsRules": {},
    "rules": {},
    "rulesDirectory": []
}
```

#### Actual behavior
There is a missing semicolon error on the line isName(this: string, name: string): boolean {
but the this parameter syntax is correct according the documentation https://www.typescriptlang.org/docs/handbook/functions.html section this parameter

#### Expected behavior
There should not be an error with correct syntax

**Comment by ajafff** [comment]
I can't reproduce this locally. Can you try tslint@5.2.0 to see if it is fixed in a recent release

**Comment by sandrozbinden** [comment]
I also tried with lint@5.2.0. You can easily reproduce the issue with cloning the project https://github.com/IMSmobile/app and then disable the comment in the file https://github.com/IMSmobile/app/blob/master/src/providers/ims-service.ts on line 99

To verify run npm install and then gulp lint.

**Comment by ajafff** [comment]
I still can't reproduce the failure. I had some problems when running `gulp lint`, so I just used tslint directly: `node_modules/.bin/tslint -p tsconfig.json --type-check`
I only got failures because of unused variables and imports, but nothing related to semicolon.

**Comment by sandrozbinden** [comment]
If I run "node_modules/.bin/tslint" -p tsconfig.json --type-check I also only got failures because of unused variables and imports. 

However if I run gulp lint (after removing the comment) I get the following error. 

[15:47:23] Using gulpfile D:\masterarbeit\app\gulpfile.js
[15:47:23] Starting 'lint'...
[15:47:24] [gulp-tslint] error (semicolon) providers/ims-service.ts[99, 48]: Missing semicolon
[15:47:24] [gulp-tslint] error (semicolon) providers/ims-service.ts[99, 60]: Missing semicolon
[15:47:24] [gulp-tslint] error (semicolon) providers/ims-service.ts[99, 70]: Missing semicolon
[15:47:24] 'lint' errored after 1.69 s
[15:47:24] Error in plugin 'gulp-tslint'

Does it have anything to do with gulp-tslint. Shouldn't the output be the same.

Here the gulp config: 
```json
var gulp = require('gulp');
var tslint = require('ionic-gulp-tslint');

gulp.task('lint', function () {
  return tslint({
    src: 'src/**/*.ts',
    tslintOptions: {
      configuration: 'tslint.json'
    },
    reporter: "verbose",
    reportOptions: {
      emitError: true
    }
  });
});
```

**Comment by ajafff** [comment]
It doesn't seem to be an issue with tslint.

I don't have any experience with gulp-tslint, so I don't know what's going wrong. Maybe it uses an older version of tslint.

**Comment by sandrozbinden** [comment]
Thanks a lot for the effort. 
The dependecies of gulp tslint seems to be ok. (https://david-dm.org/panuhorsmalahti/gulp-tslint) but even afer updating I ran into the same issues. We will now consider to use tslint directly.

**Comment by sandrozbinden** [comment]
We figured out the problem. We used the deprecated ionic-gulp-tslint task. 

Sorry for all the confusion and thanks for the effort.
